### PR TITLE
Fix memcpy kw

### PR DESCRIPTION
**Pull request by intelmark**
Changed call from memcpy to memcpy_s

Signed-off-by: Mark Feldman <mark.feldman@intel.com>

### CODE REVIEW

**Comment by tianjunwork** [comment]
Please fix build error:
CMakeFiles/SvtHevcEncApp.dir/EbAppProcessCmd.c.o: In function `ProcessInputBuffer':
EbAppProcessCmd.c:(.text+0x2826): undefined reference to `memcpy_s'

File: Source/App/EbAppProcessCmd.c [hunk 1]
```diff
@@ -825,9 +825,9 @@ static void ReadInputFrames(
                 if (config->y4m_input == EB_FALSE && config->processedFrameCount == 0 && config->inputFile == stdin) {
                     /* if not a y4m file and input is read from stdin, 9 bytes were already read when checking
                        or the YUV4MPEG2 string in the stream, so copy those bytes over */
-                    memcpy(ebInputPtr, config->y4m_buf, YUV4MPEG2_IND_SIZE);
+                    for (int i = 0; i < YUV4MPEG2_IND_SIZE; i++)
+                        *ebInputPtr++ = config->y4m_buf[0];
```
Comment (tianjunwork) [inline Source/App/EbAppProcessCmd.c]: use EB_MEMCPY

**Comment by tianjunwork** [comment]
Hi Mark, do you have any other kw issue of y4m module? Could you combine them into one PR if any? Thx.

**Comment by intelmark** [comment]
This is the remaining one.

File: Source/App/EbAppProcessCmd.c (hunk 1 above)
Comment (intelmark) [reply Source/App/EbAppProcessCmd.c]: EB_MEMCPY is defined in EBDefinitions.h in the SvtHevcEnc folder which is not in the include path.

Should I attempt to duplicate the code in perhaps in EbApi.h (in the SvtHevcEvncApp folder)



FORCE_INLINE void eb_memcpy_small(void* dst_ptr, void const* src_ptr, size_t size)
{
    const char* src = (const char*)src_ptr;
    char*       dst = (char*)dst_ptr;
    size_t      i = 0;

#ifdef _INTEL_COMPILER
#pragma unroll
#endif
    while ((i + 16) <= size)
    {
        _mm_storeu_ps((float*)(dst + i), _mm_loadu_ps((const float*)(src + i)));
        i += 16;
    }

    if ((i + 8) <= size)
    {
        _mm_store_sd((double*)(dst + i), _mm_load_sd((const double*)(src + i)));
        i += 8;
    }

    for (; i < size; ++i)
        dst[i] = src[i];
}

FORCE_INLINE void eb_memcpy_SSE(void* dst_ptr, void const* src_ptr, size_t size)
{
    const char* src = (const char*)src_ptr;
    char*       dst = (char*)dst_ptr;
    size_t      i = 0;
    size_t align_cnt = EB_MIN((64 - ((size_t)dst & 63)), size);


    // align dest to a $line
    if (align_cnt != 64)
    {
        eb_memcpy_small(dst, src, align_cnt);
        dst += align_cnt;
        src += align_cnt;
        size -= align_cnt;
    }

    // copy a $line at a time
    // dst aligned to a $line
    size_t cline_cnt = (size & ~(size_t)63);
    for (i = 0; i < cline_cnt; i += 64)
    {

        __m128 c0 = _mm_loadu_ps((const float*)(src + i));
        __m128 c1 = _mm_loadu_ps((const float*)(src + i + sizeof(c0)));
        __m128 c2 = _mm_loadu_ps((const float*)(src + i + sizeof(c0) * 2));
        __m128 c3 = _mm_loadu_ps((const float*)(src + i + sizeof(c0) * 3));

        _mm_storeu_ps((float*)(dst + i), c0);
        _mm_storeu_ps((float*)(dst + i + sizeof(c0)), c1);
        _mm_storeu_ps((float*)(dst + i + sizeof(c0) * 2), c2);
        _mm_storeu_ps((float*)(dst + i + sizeof(c0) * 3), c3);

    }

    // copy the remainder
    if (i < size)
        eb_memcpy_small(dst + i, src + i, size - i);
}

FORCE_INLINE void eb_memcpy(void  *dstPtr, void  *srcPtr, size_t size)
{
    if (size > 64) {
        eb_memcpy_SSE(dstPtr, srcPtr, size);
    }
    else
    {
        eb_memcpy_small(dstPtr, srcPtr, size);
    }
}

#define EB_MEMCPY(dst, src, size) \
	eb_memcpy(dst, src, size)

**Comment by intelmark** [comment]
I can revert back to using memcpy, or use the original fix:

for (int i = 0; i < YUV4MPEG2_IND_SIZE; i++)
                        *ebInputPtr++ = config->y4m_buf[0];

Using EB_MEMCPY might be messy.

**Comment by tianjunwork** [comment]
@intelmark, if this PR is not needed, could you close it? Thx.

File: Source/App/EbAppProcessCmd.c [hunk 2]
```diff
@@ -826,8 +826,8 @@ static void ReadInputFrames(
                     /* if not a y4m file and input is read from stdin, 9 bytes were already read when checking
                        or the YUV4MPEG2 string in the stream, so copy those bytes over */
                     memcpy(ebInputPtr, config->y4m_buf, YUV4MPEG2_IND_SIZE);
-                    headerPtr->nFilledLen += YUV4MPEG2_IND_SIZE;
                     ebInputPtr += YUV4MPEG2_IND_SIZE;
+                    headerPtr->nFilledLen += YUV4MPEG2_IND_SIZE;
```
Comment (tianjunwork) [inline Source/App/EbAppProcessCmd.c]: Do we still need this PR?

Comment (intelmark) [reply Source/App/EbAppProcessCmd.c]: I'll withdraw it - Klockwork will flag the memcpy usage, but I'll defer that issue.

-mark

**Pull request closed without merging**